addons/survey/static/src/js/survey_form.js

```diff
@@ -441,9 +441,20 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
         self.$(selectorsToFadeout.join(',')).fadeOut(this.fadeInOutDelay, function () {
             resolveFadeOut();
         });
+        // Background management - Fade in / out on each transition
+        $('div.o_survey_background').addClass('o_survey_background_transition');
 
         Promise.all([fadeOutPromise, nextScreenPromise]).then(function (results) {
-            return self._onNextScreenDone(results[1], options);
+            var nextScreenResult = results[1];
+            // always refresh, even if same background. If same, it will already be in cache anyway.
+            var nextQuestionBackground = nextScreenResult.background_image_url;
+            var preloadBackgroundPromise = self._preloadBackground(nextQuestionBackground);
+
+            return Promise.resolve(preloadBackgroundPromise).then(function (backgroundResults) {
```

Comment:
Couldn't we "await" then?
Since there is no parallel processing?

Reply:
Actually the fact that we wait here before fading might be a bit bad.
The immediate fading out was a good feedback when you clicked on the "next button".
But now if the call takes a few seconds, nothing happens on the screen?

Reply:
Can't we instead make it so the "nextQuestionPromise" only resolve when the next question is loaded AND when the image is pre-loaded?
That way we still fade out instantly?